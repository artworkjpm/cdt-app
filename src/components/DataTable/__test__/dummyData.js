export const dummyDataDesigns = [
	{
		id: 20,
		name: "20th Design",
		courses: 111,
		wales: 333,
		last_updated: "12/04/2021",
		by: "WD",
	},
];

export const dummyDataSetouts = [
	{
		courses: 200,
		id: 1,
		last_updated: "24/03/2021",
		machine_name: "RSJ_4_1",
		machine_width: 130,
		name: "1st Setout",
	},
];
