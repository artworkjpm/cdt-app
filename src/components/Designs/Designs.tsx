import axios from "axios";
import { useEffect, useState } from "react";
import { useDispatch, useSelector } from "react-redux";
import { DesignsItems, NewDesignArrayObject } from "../../models/interfaces";
import { fetchDesigns, updateAmount } from "../../redux/actions/designActions";
import DataTable from "../DataTable/DataTable";
import InfiniteScroll from "react-infinite-scroll-component";

const Designs = () => {
	const dispatch = useDispatch();
	const data = useSelector((state: { designsReducer: { designs: [DesignsItems] } }) => state.designsReducer.designs);
	const until = useSelector((state: { designsReducer: { until: number } }) => state.designsReducer.until);
	const [openDialog, setOpenDialog] = useState(false);

	useEffect(() => {
		dispatch(fetchDesigns(10));
	}, [dispatch]);

	const tableHeaders = ["Name", "Courses", "Wales", "Last_Updated", "By"];
	const editableFields = ["name", "courses", "wales"];

	function handleScroll() {
		dispatch(updateAmount(until + 10));
		dispatch(fetchDesigns(until + 10));
	}

	function handleOpen() {
		setOpenDialog(true);
	}
	function handleClose() {
		setOpenDialog(false);
	}
	const onSubmit = (editedItem: NewDesignArrayObject) => {
		axios
			.put(`http://localhost:5000/designs/${editedItem.id}`, editedItem)
			.then(() => {
				dispatch(fetchDesigns(until));
				setOpenDialog(false);
			})
			.catch((error) => {
				console.error(error);
				setOpenDialog(false);
			});
	};

	return (
		<div className="scroll-div">
			<InfiniteScroll dataLength={data.length} next={handleScroll} hasMore={true} loader="">
				<DataTable editTitle="Design" tableHeaders={tableHeaders} data={data} editableFields={editableFields} onEditSubmit={onSubmit} handleOpen={handleOpen} handleClose={handleClose} openDialog={openDialog}></DataTable>
			</InfiniteScroll>
		</div>
	);
};

export default Designs;
