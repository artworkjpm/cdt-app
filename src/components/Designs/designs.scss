.scroll-div {
	height: 60vh;
	overflow: scroll;
}
