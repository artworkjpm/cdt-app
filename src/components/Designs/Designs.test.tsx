import { mount } from "enzyme";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
import { dummyDataDesigns } from "../DataTable/__test__/dummyData";
import Designs from "./Designs";
const mockStore = configureMockStore([thunk]);

describe("<Designs>", () => {
	it("should render a startup component if startup is not complete", () => {
		const store = mockStore({
			designsReducer: { designs: dummyDataDesigns },
		});
		const wrapper = mount(
			<Provider store={store}>
				<Designs />
			</Provider>
		);
		expect(wrapper.debug()).toMatchSnapshot();
	});
});
