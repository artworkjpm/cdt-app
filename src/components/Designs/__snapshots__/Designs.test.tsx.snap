// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Designs> should render a startup component if startup is not complete 1`] = `
"<Provider store={{...}}>
  <Designs>
    <div className=\\"scroll-div\\">
      <InfiniteScroll dataLength={1} next={[Function: handleScroll]} hasMore={true} loader=\\"\\">
        <div style={{...}} className=\\"infinite-scroll-component__outerdiv\\">
          <div className=\\"infinite-scroll-component \\" style={{...}}>
            <DataTable editTitle=\\"Design\\" tableHeaders={{...}} data={{...}} editableFields={{...}} onEditSubmit={[Function: onSubmit]} handleOpen={[Function: handleOpen]} handleClose={[Function: handleClose]} openDialog={false}>
              <div id=\\"datatable\\">
                <table>
                  <thead>
                    <tr>
                      <th>
                        Name
                      </th>
                      <th>
                        Courses
                      </th>
                      <th>
                        Wales
                      </th>
                      <th>
                        Last Updated
                      </th>
                      <th>
                        By
                      </th>
                    </tr>
                  </thead>
                  <tbody>
                    <tr onClick={[Function: onClick]}>
                      <td>
                        20th Design
                      </td>
                      <td>
                        111
                      </td>
                      <td>
                        333
                      </td>
                      <td>
                        12/04/2021
                      </td>
                      <td>
                        WD
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </DataTable>
          </div>
        </div>
      </InfiniteScroll>
    </div>
  </Designs>
</Provider>"
`;
